### PR TITLE
Pla 1058 fix andor assess critical vulnerabilities in scala

### PR DESCRIPTION
**Pull request by JamesHeppell**

### CODE REVIEW

File: modules/lobby/src/main/SeekApi.scala [hunk 1]
```diff
@@ -59,7 +59,9 @@ final class SeekApi(
       .foldLeft(List.empty[Seek] -> Set.empty[String]) {
         case ((res, h), seek) if seek.user.id == user.id => (seek :: res, h)
         case ((res, h), seek) =>
-          val seekH = List(seek.variant, seek.daysPerTurn, seek.mode, seek.playerIndex, seek.user.id) mkString ","
+          val seekH =
+            List(Some(seek.variant.toString()), seek.daysPerTurn.map(_.toString()), Some(seek.mode.toString()), Some(seek.playerIndex), Some(seek.user.id))
```
Comment (lakinwecker) [inline modules/lobby/src/main/SeekApi.scala:63]: This seems unnecessary, why all the Some()?

File: modules/report/src/main/ReportForm.scala [hunk 2]
```diff
@@ -45,7 +45,7 @@ final private[report] class ReportForm(
         gameId = gameId,
         move = move
       )
-    })(_.export.some).verifying(captchaFailMessage, validateCaptcha _).verifying(cheatLinkConstraint)
+    })(_.values.some).verifying(captchaFailMessage, validateCaptcha _).verifying(cheatLinkConstraint)
```
Comment (lakinwecker) [inline modules/report/src/main/ReportForm.scala:48]: Was this name change necessary?

File: modules/common/src/main/WorkQueue.scala [hunk 3]
```diff
@@ -49,8 +49,12 @@ final class WorkQueue(buffer: Int, timeout: FiniteDuration, name: String, parall
           case e: TimeoutException =>
             lila.mon.workQueue.timeout(name).increment()
             lila.log(s"WorkQueue:$name").warn(s"task timed out after $timeout", e)
+            promise.failure(e)
+            (task, promise): TaskWithPromise[_]
           case e: Exception =>
             lila.log(s"WorkQueue:$name").info("task failed", e)
+            promise.failure(e)
```
Comment (lakinwecker) [inline modules/common/src/main/WorkQueue.scala:56]: I am very suspicious of this change. Why is it necessary? what even is this syntax?

File: modules/report/src/main/ReportForm.scala (hunk 2 above)
Comment (JamesHeppell) [reply modules/report/src/main/ReportForm.scala:48]: export is becoming a keyword and would have required export in below def so just changed it. Lichess also use values here.

File: modules/lobby/src/main/SeekApi.scala (hunk 1 above)
Comment (JamesHeppell) [reply modules/lobby/src/main/SeekApi.scala:63]: daysPerTurn is an Option, but the rest are not. I had to convert to string to keep compiler happy, and keep objects inside list all the same type. If dayPerTurn is None then i dont want it in the list. 
I could have used fold but that seems like more lines of code e.g. 
```
seek.daysPerTurn.fold(List(seek.variant.toString(), seek.mode.toString(), seek.playerIndex, seek.user.id))(List(seek.variant.toString(), _.toString(), seek.mode.toString(), seek.playerIndex, seek.user.id)).mkString(",")
```

File: modules/common/src/main/WorkQueue.scala (hunk 3 above)
Comment (JamesHeppell) [reply modules/common/src/main/WorkQueue.scala]: resolved in below commit.

**Pull request merged**